File: src/openfl/external/ExternalInterface.nim
type
  ExternalInterface* = ref object

var marshallExceptions*: bool = false
var available*: bool = false # Browser dependent

proc addCallback*(functionName: string, closure: proc (args: seq[RootRef]): RootRef) =
  # TODO: implement for JS target or native bridge
  discard

proc call*(functionName: string, args: varargs[RootRef]): RootRef =
  # TODO: implement for JS target or native bridge
  return nil

File: kfl/src/main/kotlin/openfl/external/ExternalInterface.kt
package openfl.external

object ExternalInterface {
    var available: Boolean = false
        private set
    var marshallExceptions: Boolean = false

    fun addCallback(functionName: String, closure: (Array<Any?>) -> Any?) {
        // ... implementation
    }

    fun call(functionName: String, vararg arguments: Any?): Any? {
        return null
    }
}
